### PR TITLE
Use `int64_t` for many things, and some other minor cleanups

### PR DESCRIPTION
**Pull request by zasdfgbnm**
We have been using many different integer types, and it is annoying because we have to cast. Let's unify it and use `int64_t` everywhere.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build --diff

**Comment by zasdfgbnm** [comment]
!build --diff

File: csrc/tensor_view.cpp [hunk 1]
```diff
@@ -45,136 +45,6 @@ TensorView::TensorView(
       domain_(domain),
       memory_type_(mtype) {}
 
-TensorView::TensorView(
```
Comment (zasdfgbnm) [inline csrc/tensor_view.cpp:48]: Note on non `int64_t` changes

File: csrc/utils.h [hunk 2]
```diff
@@ -551,4 +551,17 @@ V getOrDefault(const std::unordered_map<K, V>& map, const K& key) {
   return i == map.end() ? V() : i->second;
 }
 
+inline int64_t wrapDim(int64_t dim, int64_t ndim) {
```
Comment (zasdfgbnm) [inline csrc/utils.h:554]: Note on non-int64 changes. We had be doing this manually many times, so I created a utility function for it.

File: tests/cpp/test_gpu1.cpp [hunk 3]
```diff
@@ -547,70 +537,6 @@ TEST_F(NVFuserTest, FusionTopoSort_CUDA) {
   NVF_CHECK(v6->definition()->name() == 3);
 }
 
-TEST_F(NVFuserTest, FusionTensor_CUDA) {
```
Comment (zasdfgbnm) [inline tests/cpp/test_gpu1.cpp:550]: Note: test removed

**Comment by zasdfgbnm** [comment]
!build --diff

**Comment by zasdfgbnm** [comment]
!build --diff

**Comment by zasdfgbnm** [comment]
!build --diff

**Comment by zasdfgbnm** [comment]
!build --diff

**Comment by zasdfgbnm** [comment]
!build --diff

**Comment by zasdfgbnm** [comment]
Looks like I don't need to worry about jit_python_bc_advisory_17_A100?

File: csrc/ir/utils.cpp [hunk 4]
```diff
@@ -250,9 +249,9 @@ Expr* transferDefinitionToNewOutputs(
   return mutator.mutateExprOutputsOnly(expr);
 }
 
-TensorView* rfactorHelper(
+TensorView* rFactorHelper(
```
Comment (jjsjann123) [inline csrc/ir/utils.cpp:252]: nitpicking: is `rfactor` not a word? It reads a bit strange.

File: csrc/ir/interface_nodes.h [hunk 5]
```diff
@@ -550,33 +546,30 @@ class NVF_API TensorView : public Val {
...
-    }
-    return pos;
+  int64_t wrapDim(int64_t dim) const {
+    return nvfuser::wrapDim(dim, nDims());
   }
 
   //! A helper function to maintain the consistency of schedules of
   //! multiple outputs wheen doing rfactor on multi-output reduction ops.
-  TensorView* multiOutputRfactorHelper(
+  TensorView* multiOutputrFactorHelper(
```
Comment (jjsjann123) [inline csrc/ir/interface_nodes.h:555]: so I think `RFactor` should be the consistent one instead.

File: csrc/ir/utils.cpp (hunk 4 above)
Comment (jjsjann123) [reply csrc/ir/utils.cpp:252]: nvm, I see where you are going with this.

**Review by jjsjann123 — COMMENTED** [review]
:clap:

File: csrc/ir/interface_nodes.h (hunk 5 above)
Comment (zasdfgbnm) [reply csrc/ir/interface_nodes.h]: Changed

File: csrc/ir/utils.cpp (hunk 4 above)
Comment (zasdfgbnm) [reply csrc/ir/utils.cpp:252]: Yeah, I think rFactor corresponds to reduction-factor, which is a compound word.

Comment (jjsjann123) [reply csrc/ir/utils.cpp:252]: ^^^ yeah, but we are using `xxx_rfactor_yyy` in some variable names, which sorta blend them as a single word.

Not saying that I'm in favor of `xxx_r_factor_yyy`. :laughing: I'm happy with your clean up already.

**Review by jjsjann123 — APPROVED** [review]
:shipit:

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
^^^ Just in case the renaming might step on someone else's PRs.

**Comment by zasdfgbnm** [comment]
!build

**Pull request merged**